pkg/client/openstack/scoped/logging.go

```diff
@@ -0,0 +1,43 @@
+package scoped
+
+import (
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/gophercloud/gophercloud/openstack/identity/v3/tokens"
+
+	"github.com/sapcc/kubernikus/pkg/api/models"
+)
+
+type LoggingClient struct {
+	Client Client
+	Logger log.Logger
+}
+
+func (c LoggingClient) GetMetadata() (metadata *models.OpenstackMetadata, err error) {
+	defer func(begin time.Time) {
+		c.Logger.Log(
+			"msg", "fetched metadata",
+			"flavors", len(metadata.Flavors),
+			"keypairs", len(metadata.KeyPairs),
+			"routers", len(metadata.Routers),
+			"security_groups", len(metadata.SecurityGroups),
+			"took", time.Since(begin),
+			"err", err,
+		)
+	}(time.Now())
+
+	return c.Client.GetMetadata()
+}
+
+func (c LoggingClient) Authenticate(authOptions *tokens.AuthOptions) (err error) {
+	defer func(begin time.Time) {
+		c.Logger.Log(
+			"msg", "authenticated",
+			"took", time.Since(begin),
+			"v", 2,
```

Comment:
I'm wondering if we should log authentication errors at a higher level then successful auth attempts. Not sure what else is logged but when openstack auth is failing we somehow see this without enabling `verbose` logging.

Reply:
done for this specific instance